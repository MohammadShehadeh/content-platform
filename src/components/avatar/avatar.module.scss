.avatar {
  display: flex;
  align-items: center;
  justify-content: center;
  padding: var(--space-md);
  border-radius: 50%;
  background-color: hsl(var(--primary) / 0.1);
  color: hsl(var(--primary));
}
