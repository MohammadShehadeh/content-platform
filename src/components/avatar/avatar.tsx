import { UserIcon } from 'lucide-react';

import { cn } from '@/lib/utils';

import styles from './avatar.module.scss';

interface AvatarProps {
  size?: number;
  className?: string;
  children?: React.ReactNode;
}

export const Avatar = ({ size = 32, className }: AvatarProps) => {
  return (
    <div className={cn(styles.avatar, className)} data-testid="avatar">
      <UserIcon size={size} />
    </div>
  );
};
