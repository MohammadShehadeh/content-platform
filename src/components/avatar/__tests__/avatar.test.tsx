import { render, screen } from '@testing-library/react';

import { Avatar } from '@/components/avatar';

describe('Avatar', () => {
  it('should render with custom className', () => {
    render(<Avatar className="custom-avatar" />);
    const avatarElement = screen.getByTestId('avatar');
    expect(avatarElement).toHaveClass('custom-avatar');
  });

  it('should match snapshot', () => {
    const { container } = render(<Avatar />);
    expect(container).toMatchSnapshot();
  });
});
