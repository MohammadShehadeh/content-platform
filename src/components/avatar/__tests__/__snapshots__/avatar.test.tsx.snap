// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar matches snapshot 1`] = `
<div>
  <div
    class="avatar"
    data-testid="avatar"
  >
    <svg
      class="lucide lucide-user"
      fill="none"
      height="32"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="32"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M19 21v-2a4 4 0 0 0-4-4H9a4 4 0 0 0-4 4v2"
      />
      <circle
        cx="12"
        cy="7"
        r="4"
      />
    </svg>
  </div>
</div>
`;

exports[`Avatar should match snapshot 1`] = `
<div>
  <div
    class="avatar"
    data-testid="avatar"
  >
    <svg
      class="lucide lucide-user"
      fill="none"
      height="32"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="32"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M19 21v-2a4 4 0 0 0-4-4H9a4 4 0 0 0-4 4v2"
      />
      <circle
        cx="12"
        cy="7"
        r="4"
      />
    </svg>
  </div>
</div>
`;
